docs/guides/go-built-in.md

```diff
@@ -457,30 +465,15 @@ The application is free to modify the group before returning from the call, as l
 does not use more bytes than `RequestPrepareProposal.max_tx_bytes`
 For example, the application may reorder, add, or even remove transactions from the group to improve the
 execution of the block once accepted.
+
 In the following code, the application simply returns the unmodified group of transactions:
 
 ```go
- func (app *KVStoreApplication) PrepareProposal(_ context.Context, proposal *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
-   totalBytes := int64(0)
-   txs := make([]byte, 0)
-
-   for _, tx := range proposal.Txs {
-     totalBytes += int64(len(tx))
-     txs = append(txs, tx...)
-
-       if totalBytes > int64(proposal.MaxTxBytes) {
-         break
-       }
-     }
-
-     return &abcitypes.ResponsePrepareProposal{Txs: proposal.Txs}, nil
- }
+func (app *KVStoreApplication) PrepareProposal(_ context.Context, proposal *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
+    return &abcitypes.ResponsePrepareProposal{Txs: proposal.Txs}, nil
+}
```

Comment:
It is possibly now that the list of transactions returned is bigger that what fits in the block. So the app is supposed to reduce the list of transactions to fit in the block.
Given that this is an intro document, I am fine with the simpler document, but wonder if we should have a warning here.

Reply:
> It is likely now that the list of transactions returned is bigger than what fits in the block.

Could you please explain how? CometBFT checks the size when reaping transactions from the mempool, right? So naturally, if the application does not modify the proposed transactions list, then it can safely skip the check, no?

Reply:
This behavior was changed to allow reaping all the txs. See the following excerpt of https://docs.cometbft.com/v0.38/spec/abci/abci++_methods#prepareproposal

"The Application MAY configure CometBFT to include a list of transactions in RequestPrepareProposal.txs whose total size in bytes exceeds RequestPrepareProposal.max_tx_bytes. If the Application sets ConsensusParams.Block.MaxBytes to -1, CometBFT will include all transactions currently in the mempool in RequestPrepareProposal.txs, which may not fit in RequestPrepareProposal.max_tx_bytes. Therefore, if the size of RequestPrepareProposal.txs is greater than RequestPrepareProposal.max_tx_bytes, the Application MUST remove transactions to ensure that the RequestPrepareProposal.max_tx_bytes limit is respected by those transactions returned in ResponsePrepareProposal.txs. "

Reply:
I didn't know that! I believe we all agree that the check shouldn't be in an introduction guide because the **default for `MaxBytes` is 4MB, not `-1`**. I will open a separate issue with documentation improvements and include this item. Thanks again 🙏